OpenRA.Game/Graphics/SoftwareCursor.cs

```diff
@@ -78,27 +82,30 @@ public void Render(Renderer renderer)
 				return;
 
 			var cursorSequence = cursorProvider.GetCursorSequence(cursorName);
-			var cursorSprite = sprites[cursorName][Frame];
+			var cursorSprite = sprites[cursorName][(int)cursorFrame % cursorSequence.Length];
 			var cursorSize = CursorProvider.CursorViewportZoomed ? 2.0f * cursorSprite.Size : cursorSprite.Size;
 
 			var cursorOffset = CursorProvider.CursorViewportZoomed ?
 				(2 * cursorSequence.Hotspot) + cursorSprite.Size.XY.ToInt2() :
 				cursorSequence.Hotspot + (0.5f * cursorSprite.Size.XY).ToInt2();
 
 			renderer.SetPalette(palette);
+			var mousePos = isLocked ? lockedPosition : Viewport.LastMousePos;
 			renderer.SpriteRenderer.DrawSprite(cursorSprite,
-				Viewport.LastMousePos - cursorOffset,
+				mousePos - cursorOffset,
 				paletteReferences[cursorSequence.Palette],
 				cursorSize);
 		}
 
-		public int Frame
+		public void Lock()
 		{
-			get
-			{
-				var cursorSequence = cursorProvider.GetCursorSequence(cursorName);
-				return (int)cursorFrame % cursorSequence.Length;
-			}
+			lockedPosition = Viewport.LastMousePos;
+			isLocked = true;
```

Comment:
We'll need to call `Game.Renderer.Window.SetRelativeMouseMode(true);` here too so that the cursor will return to the mousedown location on mouseup.